### PR TITLE
CB-88: Improve Search by Concept/Observation

### PR DESCRIPTION
**Pull request by ghost**
# JIRA TICKET NAME:
CB-88:[ Improve Search by Concept/Observation](https://issues.openmrs.org/browse/CB-88)

## SUMMARY:
This PR Improves and refactors concept search, by giving users the ability to make queries base on concepts observations.

### CODE REVIEW

**Comment by andela-aolaniran** [comment]
Nice one with the auto-complete search field @andela-emurphy .
I noticed the search labels don't match what is selected in the search form. 
Example is what is shown in the two searches in the snapshot. They are different searches with different result sets but their labels are the same:
image

**Comment by dkayiwa** [comment]
Thanks @andela-aolaniran for reviewing this. Keep it up. 👍

**Comment by andela-aolaniran** [comment]
@dkayiwa Yea, @andela-emurphy  actually asked that I pulled his branch and test it so we could be sure apart from the changes in code.
I think it should be possible to have the branch deploy it's self on PRs so the changes can be tested in a staging/testing environment before they are merged. Or, we could manually upload the changes to modules-refapp.openmrs.org/ so they can be tested.

**Comment by dkayiwa** [comment]
If you test them locally and find them to be working well, then the code is ripe for a pull request. ☺️

File: app/js/components/tabs/tabcomponents/observationComponent.jsx [hunk 1]
```diff
@@ -1,165 +1,381 @@
...
-    handleDateChange(dateType) {
-        return value => this.setState({
-            [dateType]: this.getDateString(value)
+    setRef(input) {
+        this.textInput = input; 
+    }
+
+    handleSubmit(event) {
+        event.preventDefault();
+        const types = {
```
Comment (ghost) [inline app/js/components/tabs/tabcomponents/observationComponent.jsx:54]: Base on the datatype of the concepts, I wrote The handle submit method to dynamically build the query using the appropriate library key. 

cc @dkayiwa

Comment (dkayiwa) [reply app/js/components/tabs/tabcomponents/observationComponent.jsx:54]: Do you mind pointing me to where you do that in the cohort builder?

Comment (ghost) [reply app/js/components/tabs/tabcomponents/observationComponent.jsx:54]: Link to the observation component [click here](https://github.com/andela-odaniel/openmrs-owa-cohortbuilder/blob/118097f6aa05966e72d0164b53e58a3b8c2f95b1/app/js/components/tabs/tabcomponents/observationComponent.jsx)

screen shot 2017-07-03 at 9 53 58 pm

**Comment by dkayiwa** [comment]
@andela-aolaniran and @andela-emurphy do you plan to add any more commits to this pull request?

**Comment by ghost** [comment]
Yes @dkayiwa

**Comment by dkayiwa** [comment]
@andela-emurphy which means it is not ready for merging. Correct?

**Comment by ghost** [comment]
@andela-aolaniran noticed a minor issue with the PR that I'm yet to fix. I won't be available today since we have an outing at work. I'll fix it by tomorrow and get back to you.

**Comment by enahomurphy** [comment]
@dkayiwa  I have fixed the issue @andela-aolaniran raised.

**Comment by dkayiwa** [comment]
@enahomurphy that is awesome! 😊

**Comment by dkayiwa** [comment]
@andela-emurphy @andela-aolaniran now that i have merged, can we update modules-refapp with the latest changes?

**Comment by andela-aolaniran** [comment]
@dkayiwa Sure thing. The modules-refapp has been updated with the latest changes.

**Comment by dkayiwa** [comment]
@andela-aolaniran can you also respond to the talk thread and point Ellen to it?

**Comment by dkayiwa** [comment]
@andela-aolaniran can you also include the cohort builder app on the home page of that server?

**Pull request merged**